### ISSUE TITLE
Does it Read .bashrc?

### ISSUE THREAD

**Issue by brmullikin** [chrissimpkins/glue#12]
I wonder if glue supports .bashrc (or .bash_profile on mac)?

Perhaps I need to set the path somewhere?

I didn't see anything in the documentation about it, so I thought I would ask.

Cheers, and thanks for the wonderful plugin!

**Comment by N4M3Z** [comment]
+1

**Comment by dnut** [comment]
This is actually pretty easy to implement as a user. Of course, one way would be to convert your aliases and functions to json, which is also easy, but that introduces the problem of having to maintain two separate files, so it's far from ideal. If you just want to use your bashrc file in Glue, complete these steps:

- Convert your aliases to functions, which is pretty straightforward, especially in Sublime (take advantage of alt-F3). Aliases and functions handle recursion differently, so keep some as aliases to prevent infinite loops, such as things like ```alias ls='ls --color=auto'```. The things you keep as aliases will not work in Glue. The following would be an appropriate substitution:
```shell
alias foo="echo 'bar'"    # remove
foo() { echo 'bar'; }     # add
```
- Make sure your ~/.bashrc works regardless of whether it's running interactively. I had to comment out a line I had on top of the file:
```shell
[[ $- != *i* ]] && return
```
- Following [these instructions](https://gluedocs.readthedocs.org/en/latest/extend-glue.html), create a Glue alias that sources your bashrc file. Here is what I used:
```json
"rc": ". $HOME/.bashrc; {{args}}"
```

Whenever you want to call a function from your bashrc in Glue, use a command like this:
```
glue rc [command]
```
Don't expect everything to work perfectly, but most things should be fine. If you have something that requires user input, get creative.
